Rules/Sandbox/Scripts/Sandbox_Rules.as

```diff
@@ -378,5 +386,41 @@ void Reset(CRules@ this)
 	this.set("core", @core);
 	this.set("start_gametime", getGameTime() + core.warmUpTime);
 	this.set_u32("game_end_time", getGameTime() + core.gameDuration); //for TimeToEnd.as
+
+	kegCount = 0;
+	mineCount = 0;
+}
+
+void onBlobCreated(CRules@ this, CBlob@ blob)
+{
+	if (blob.getName() == "mine")
+	{
+		mineCount += 1;
+		if (mineCount > maxMines)
+		{
+			blob.server_Die(); // wont explode because its just been made
+		}
+	}
+	else if (blob.getName() == "keg")
+	{
+		kegCount += 1;
+		if (kegCount > maxKegs)
+		{
+			blob.server_Die();
```

Comment:
Wouldn't it make more sense if the oldest keg is removed, not the most recent? Couldn't someone just hide a lot of kegs underground somewhere so it's impossible for someone else to spawn new ones?

This also applies to mines too.